dali/kernels/audio/mel_scale/mel_scale.h

```diff
@@ -70,6 +72,88 @@ struct SlaneyMelScale {
   }
 };
 
+template <typename T, int Dims>
+class MelFilterImplBase {
+ public:
+  template <typename MelScale>
+  MelFilterImplBase(MelScale mel_scale, const MelFilterBankArgs &args): args_(args) {
+    assert(args.sample_rate > 0);
+
+    assert(args.freq_low >= 0 && args.freq_low <= args.sample_rate / 2);
+    mel_low_ = mel_scale.hz_to_mel(args.freq_low);
+
+    assert(args.freq_high >= 0 && args.freq_high <= args.sample_rate / 2);
+    mel_high_ = mel_scale.hz_to_mel(args.freq_high);
+
+    int nfilter = args.nfilter;
+    assert(nfilter > 0);
+
+    int64_t nfft = args.nfft;
+    assert(nfft > 0);
+
+    hz_step_ = args.sample_rate / nfft;
+    mel_delta_ = (mel_high_ - mel_low_) / (nfilter + 1);
+
+    fftbin_size_ = nfft / 2 + 1;
+    double inv_hz_step = 1.0f / hz_step_;
+    fftbin_start_ = std::ceil(args.freq_low * inv_hz_step);
+    if (fftbin_start_ < 0)
+      fftbin_start_ = 0;
+    fftbin_end_ = std::floor(args.freq_high * inv_hz_step);
+    if (fftbin_end_ > fftbin_size_ - 1)
+      fftbin_end_ = fftbin_size_ - 1;
+
+    weights_down_.resize(fftbin_size_);
+    norm_factors_.resize(nfilter, T(1));
+    T mel0 = mel_low_, mel1 = mel_low_ + mel_delta_;
```

Comment:
double? are mel_low / mel_high used in some long loop? If not -> double

Reply:
done